clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1075,6 +1078,7 @@ void invokeCompletedOffsetCommitCallbacks() {
     public RequestFuture<Void> commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, final OffsetCommitCallback callback) {
         invokeCompletedOffsetCommitCallbacks();
 
+        inFlightAsyncCommits.incrementAndGet();
```

Comment:
should we move this to the top of `sendOffsetCommitRequest`? Technically if we don't have a coordinator (see the else block), then there's really no inflight request. Which means, should we also remove line 1142 as it is failed coordinator request handling.

Reply:
@philipnee I rebased the branch on trunk so the line numbers changed a bit. Instead of line 1142, Is it now line 1118 (in the onFailure block inside `commitOffsetsAsync`)?

I made changes based on your idea. I think we should keep the increment in `doCommitOffsetsAsync` because that is also where the decrement is. WDYT?

Also, instead of using a try/finally, I moved the decrement to the top of the callback method.

This is committed in https://github.com/apache/kafka/pull/13678/commits/f8ce359de5e0a7921bfac26da39feb2af11a9f09.